command/plugins_install.go

```diff
@@ -128,3 +198,167 @@ func (c *PluginsInstallCommand) RunContext(buildCtx context.Context, args []stri
 
 	return 0
 }
+
+func (c *PluginsInstallCommand) InstallFromBinary(args *PluginsInstallArgs) int {
```

Comment:
I see what you are doing here. Is there a way to reuse the Checksummer for obtaining and write the checksum file, the InFolders, and BinaryInstallOptions for generating file names. 

I have to think through this a little but would there be any benefit to adding this as a method on `plugingetter.Requirement`?

You can write the code to be something like the following and reuse a bit of the logic from InstallLatest along with the call to describe you have in InstallFromBinary. 


```
	// a plugin requirement that matches them all
	pluginRequirement := plugingetter.Requirement{
		Identifier: plugin,
	}

	if args.Version != "" {
		constraints, err := version.NewConstraint(args.Version)
		if err != nil {
			c.Ui.Error(err.Error())
			return 1
		}
		pluginRequirement.VersionConstraints = constraints
	}

	if runtime.GOOS == "windows" && opts.Ext == "" {
		opts.BinaryInstallationOptions.Ext = ".exe"
	}

         var newInstall *plugingetter.Installation
         var err error
	// If we did specify a binary to install the plugin from, we ignore
	// the Github-based getter in favour of installing it directly.
	if args.PluginPath != "" {
		newInstall, err :=  pluignRequirement.InstallLocalBinary(args.PluginPath, plugingetter.InstallOptions{
		InFolders:                 opts.FromFolders,
		BinaryInstallationOptions: opts.BinaryInstallationOptions,
		Force:                     args.Force, //This may not be needed.
	       })
	} else {

	getters := []plugingetter.Getter{
		&github.Getter{
			UserAgent: "packer-getter-github-" + pkrversion.String(),
		},
	}

	newInstall, err := pluginRequirement.InstallLatest(plugingetter.InstallOptions{
		InFolders:                 opts.FromFolders,
		BinaryInstallationOptions: opts.BinaryInstallationOptions,
		Getters:                   getters,
		Force:                     args.Force,
	})
     }
....
```